### ISSUE TITLE
Track the packaging progress

### ISSUE THREAD

**Issue by ocefpaf** [ODM2/YODA-Tools#20]
@sreeder and @emiliom this issue is to help me track what I need to do here. Please let me know what you think.

- [ ] fix `Travis-CI` back online
- [ ] add `appveyor` testing (Windows users were mentioned in the meeting)
- [ ] add a `setup.py`
- [ ] decide on how to manage versioning (@sreeder are you OK with `versioneer` or do you want to handle that differently?)
- [ ] source (wheels?) distribution for `PyPI `
- [ ] conda recipe and package for the `ODM2` channel
- [ ] `entrypoint` to call the GUI
- [ ] "desktop app" metadata like icons, etc

**Comment by emiliom** [comment]
@ocefpaf, I like this, and in fact was going to suggest something along these lines. Thanks.

**Comment by sreeder** [comment]
@ocefpaf  `versioneer`  is great

**Comment by ocefpaf** [comment]
> @ocefpaf  `versioneer`  is great

Thanks @sreeder! As soon as we clear #21 and #22 I'll start working on the `setup.py` and  `versioneer`.

**Comment by ocefpaf** [comment]
@sreeder do you have some time this week to go over the possible `entrypoints` for  `YODA-Tools`?
Or maybe we need this one to call the GUI? That is the only missing part to finish #24

**Comment by sreeder** [comment]
@ocefpaf yes, I will take a look at it. To double check what you want for "entry points", It is basically where you can access the code? So the .py file where they send in the command line arguments? Or the GUI .py file.

**Comment by ocefpaf** [comment]
> @ocefpaf yes, I will take a look at it.

Thanks!

> To double check what you want for "entry points", It is basically where you can access the code? So the .py file where they send in the command line arguments? Or the GUI .py file.

In a way it can be both. We can have multiple entry points for CLI, for example, or 1 (or a few) for GUIs.

In this case I need the one used to call the GUI main loop. (I am guessing there there is only 1.)

**Comment by sreeder** [comment]
@ocefpaf  the file that I use to run the converter GUI is D:\DEV\YODA-Tools\yodatools\dataloader\controller\Main.py.

**Comment by ocefpaf** [comment]
> @ocefpaf  the file that I use to run the converter GUI is D:\DEV\YODA-Tools\yodatools\dataloader\controller\Main.py. 

Thanks! I'll experiment adding that to the `setup.py` tomorrow morning.

PS: we can do icon install, etc. But I don't want to get into that right now.
I'll started with only a command that opens the GUI.
